imagetest/testworkflow.go

```diff
@@ -52,6 +52,45 @@ type TestWorkflow struct {
 	counter int
 }
 
+// Network represent network used by vm in setup.go.
+type Network struct {
+	name         string
+	testWorkflow *TestWorkflow
+}
+
+// SubNetwork represent subnetwork used by vm in setup.go.
+type SubNetwork struct {
+	name         string
+	testWorkflow *TestWorkflow
+}
+
+// AddSecondaryRange add secondary IP range to SubNetwork
+func (s SubNetwork) AddSecondaryRange(rangeName, ipRange string) error {
+	createSubnetworksStep, ok := s.testWorkflow.wf.Steps[createSubNetworkStepName]
+	if !ok {
+		return fmt.Errorf("create-sub-network step missing")
+	}
+
+	subnetwork, err := s.findSubNetwork(*createSubnetworksStep.CreateSubnetworks)
+	if err != nil {
+		return err
+	}
+	subnetwork.SecondaryIpRanges = append(subnetwork.SecondaryIpRanges, &compute.SubnetworkSecondaryRange{
+		IpCidrRange: ipRange,
+		RangeName:   rangeName,
+	})
+	return nil
+}
+
+func (s SubNetwork) findSubNetwork(subnetworks daisy.CreateSubnetworks) (*daisy.Subnetwork, error) {
```

Comment:
in all other cases we do the matching loop inline: in `AddMetadata`, in `EnableSecureBoot`, in `SetCustomNetwork`, etc.

but if you really want to make this a method, don't pass in subnetworks. look it up here as you do on line 69 in AddSecondaryRange above.

Reply:
Follow the paradigm in `AddMetadata`, in` EnableSecureBoot`, in `SetCustomNetwork`. just make the `AddSecondaryRange` as the these method without return error